src/pages/login.js

```diff
@@ -5,33 +5,65 @@ import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
 import {Sae} from 'react-native-textinput-effects';
 import {observer} from 'mobx-react/native'
 
-import {routes}  from '../routes';
+import routes  from '../routes';
 import core_styles from '../styles/core-styles';
 import theme       from '../styles/ui-theme';
 import API from "../services/api";
 import User from "../models/user";
 import FacebookLoginButton from "../components/FacebookLoginButton";
 import LocalStorage from "../services/localStorage";
+import Loading from '../components/Loading';
 
 @observer
 class Login extends Component {
   static propTypes = {
     navigator: PropTypes.object.isRequired,
     route: PropTypes.object.isRequired,
+    store: PropTypes.object.isRequired,
   };
   /** @type LocalStorage */
   localStorage = LocalStorage.get();
 
+  componentWillMount() {
+    this.redirectIfLoggedIn()
+  }
+
   async logIn(user: User) {
     try {
+      await this.setSelf(user);
+      this.props.navigator.replace(routes.chat)
+    }
+    catch (e) {
+      console.error(e);
+    }
+  }
+
+  async redirectIfLoggedIn() {
```

Comment:
i was redirecting too once the user login so we need to set up the navigation flow maybe :D